avocado/core/test.py

```diff
@@ -601,15 +598,10 @@ class SimpleTest(Test):
     def __init__(self, name, params=None, base_logdir=None, tag=None, job=None):
         super(SimpleTest, self).__init__(name=name, params=params,
                                          base_logdir=base_logdir, tag=tag, job=job)
+        if not hasattr(self, 'filename'):
+            self.filename = os.path.abspath(self.name)
```

Comment:
You have to set this before calling the super class (basically just use the patch I provided, it passed the CI).